crates/context_aware_config/src/api/context/handlers.rs

```diff
@@ -526,3 +530,48 @@ async fn bulk_operations(
     })?;
     Ok(Json(response))
 }
+
+#[put("/priority/recompute")]
+async fn priority_recompute(
+    db_conn: DbConnection,
+    _user: User,
+) -> superposition::Result<HttpResponse> {
+    use crate::db::schema::contexts::dsl::*;
+    let DbConnection(mut conn) = db_conn;
+
+    let result: Vec<Context> = contexts.load(&mut conn).map_err(|err| {
+        log::error!("failed to fetch contexts with error: {}", err);
+        db_error!(err)
+    })?;
+
+    let dimension_schema_map = get_all_dimension_schema_map(&mut conn)?;
+    let update_contexts: Vec<Context> = result
+        .iter()
+        .map(|context| Context {
+            priority: calculate_context_priority(
+                "context",
+                &context.value,
+                &dimension_schema_map,
+            )
+            .unwrap(),
+            ..context.clone()
+        })
+        .collect::<Vec<Context>>();
+    let insert = diesel::insert_into(contexts)
+        .values(&update_contexts)
+        .on_conflict(id)
+        .do_update()
+        .set(priority.eq(excluded(priority)))
+        .execute(&mut conn);
+
+    match insert {
+        Ok(_) => Ok(HttpResponse::Ok()
+            .json(json!({"message": "Contexts priority updated successfully"}))),
```

Comment:
Can we send back a response like this:
```
[
    {
        context: String
        old_priority: some value
        new_priority: some value
    },
    {
        next_context
    },
    .........
]
```

Reply:
@Datron thinking why this response would be necessary, its a batch operation with many changes , giving an one time response does not make much sense here I think

Reply:
@pratikmishra356 Its useful for shell commands and to let the user know what has changed.

Reply:
done